File: Introduction to Robotics Course/Chapter5/Lesson3/Chapter5_Lesson3.m

l1 = 1.0; l2 = 0.7;
q1_min = -pi; q1_max = pi;
q2_min = -pi; q2_max = pi;

N = 50000;
q1 = q1_min + (q1_max - q1_min)*rand(N,1);
q2 = q2_min + (q2_max - q2_min)*rand(N,1);

x = l1*cos(q1) + l2*cos(q1 + q2);
y = l1*sin(q1) + l2*sin(q1 + q2);

figure; scatter(x,y,1,'.');
axis equal; grid on;
title('Estimated workspace of planar 2R arm');
xlabel('x'); ylabel('y');
      

File: Introduction_to_Robotics_Course/Chapter5/Lesson3/Chapter5_Lesson3.py

import numpy as np
import matplotlib.pyplot as plt

# Link lengths
l1, l2 = 1.0, 0.7

# Joint limits (radians)
q1_min, q1_max = -np.pi, np.pi
q2_min, q2_max = -np.pi, np.pi

N = 50000
q1 = np.random.uniform(q1_min, q1_max, N)
q2 = np.random.uniform(q2_min, q2_max, N)

# Geometric position (planar)
x = l1*np.cos(q1) + l2*np.cos(q1 + q2)
y = l1*np.sin(q1) + l2*np.sin(q1 + q2)

plt.figure()
plt.scatter(x, y, s=1)
plt.axis("equal")
plt.title("Estimated workspace of planar 2R arm")
plt.xlabel("x")
plt.ylabel("y")
plt.show()
      